src/middleware.ts

```diff
@@ -0,0 +1,19 @@
+import { type NextRequest, NextResponse } from "next/server";
+
+const hasAuthToken = (request: NextRequest) => {
+  const cookie = request.cookies;
+  const token = cookie.get("access_token");
+  return token ? false : true;
+};
+
+export default function middleware(request: NextRequest) {
+  if (hasAuthToken(request)) {
+    return NextResponse.redirect(new URL("/login", request.url));
```

Comment:
Its not clear, it's look like you are checking if user has token then you redirect it to login. you should put `!hasAuthToken(request)`